whois-nrtm4-client/src/main/java/net/ripe/db/nrtm4/client/dao/Nrtm4ClientMirrorRepository.java

```diff
@@ -49,11 +61,24 @@ SELECT id, source, MAX(version), session_id, type, created
                         rs.getString(2),
                         rs.getLong(3),
                         rs.getString(4),
-                        rs.getString(5),
+                        NrtmClientDocumentType.fromValue(rs.getString(5)),
                         rs.getLong(6)
                         ));
     }
 
+    public void persistRpslObject(final RpslObject rpslObject){
+        try {
+            final long now = JdbcRpslObjectOperations.now(dateTimeProvider);
+            jdbcMasterTemplate.update("INSERT INTO last_mirror (object, object_type, pkey, timestamp) VALUES (?, ?, ?, ?)",
+                    getRpslObjectBytes(rpslObject),
+                    ObjectTypeIds.getId(rpslObject.getType()),
+                    rpslObject.getKey().toString(),
+                    now);
+        } catch (IOException e) {
+            LOGGER.error("unable to get the bytes of the object {}", rpslObject.getKey(), e);
```

Comment:
Separate `getRpslObjectBytes` into variable and catch `IOException` separately

Reply:
I return an error as well, because if not maybe some objects are added and some of them are not...
That would generate an incomplete snapshot